code/game/machinery/vending.dm

```diff
@@ -2128,3 +2128,58 @@
 	contraband = list(/obj/item/weapon/reagent_containers/food/drinks/soda_cans/quantum = 5)
 
 	pack = /obj/structure/vendomatpack/nuka
+
+/obj/machinery/vending/mining
+	name = "Dwarven Mining Equipment"
+	desc = "Get your mining equipment here, and above all keep digging!"
+	product_slogans = "This asteroid isn't going to dig itself!;Stay safe in the tunnels, bring two Kinetic Accelerators!;Jetpacks, anyone?"
+	product_ads = "Hungry, thirsty or unequipped? We have your fix!"
+	vend_reply = "What a glorious time to mine!"
+	icon_state = "mining"
+	req_access_txt = "48"

```

Comment:
access.dm, use the vars there (yes they're not defines but they act as them, effectively)